src/cljdoc/server/system.clj

```diff
@@ -66,6 +71,15 @@
 (defmethod ig/init-key :cljdoc/dogstats [_ {:keys [uri tags]}]
   (dogstatsd/configure! uri {:tags tags}))
 
+(defmethod ig/init-key :cljdoc/cache [_ {:keys [cache-dir]}]
+  (.mkdirs (io/file cache-dir))
+  {:cljdoc.util.repositories/get-pom-xml (sqlite-cache/memo-sqlite
```

Comment:
`init-key` can also return plain functions so you could just return the memoized function making usage a little more concise in other places.

Reply:
Thats a good idea but I wanted to return a `map` because we can use this `map` to add more memoized functions in future. For ex: may be we can have something like `{... :cljdoc.util.repositories/get-github-stars (sqlite-cache/memo-sqlite ...)}` etc. 

Is this the right way to think? or am I missing something?

Reply:
I think in this case it would be more idiomatic to just add the additional caches as other top level items in the integrant system map?